pkg/controller/expectations.go

```diff
@@ -48,6 +48,8 @@ const (
 	ExpectationsTimeout = 5 * time.Minute
 )
 
+const keyFuncChangedFormat = "KeyFunc was changed, %v"
```

Comment:
nit: it's not just the format, it's actually a constant for the error message itself. Maybe different name ?